### PR TITLE
Add "random" mode

### PR DESCRIPTION
**Pull request by cszczepaniak**
In addition to the daily game, some users are just begging for more ways to play. This PR adds a "random" mode that simply creates a random puzzle based on nothing (other than the seed of `math/rand`). Users can continually get new puzzles on-demand!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ To play, follow one of the installation guides below. After that, you can just n
 You might also want to `git pull` to ensure that your git history is up-to-date. Otherwise, you may end up playing the wrong game for the day. The program doesn't do this automatically because I didn't want it to make any changes to the file system.
 
 ### Installation from source (recommended)
-With any v1.21 or greater of [Golang](https://go.dev/) you can easily install from source:
+With any v1.22 or greater of [Golang](https://go.dev/) you can easily install from source:
```
Comment (cszczepaniak) [inline README.md:14]: I _think_ this requires 1.22 because the `go.mod` specifies `go 1.22.3`

File: internal/game/authors.go [hunk 2]
```diff
@@ -2,11 +2,12 @@ package game
 
 import (
 	"fmt"
-	"github.com/josephnaberhaus/gauthordle/internal/git"
 	"math"
 	"math/rand"
 	"slices"
 	"strings"
+
+	"github.com/josephnaberhaus/gauthordle/internal/git"
```
Comment (JosephNaberhaus) [inline internal/game/authors.go:10]: Someone has `gosimports` installed. It does look so much better, doesn't it.

File: README.md (hunk 1 above)
Comment (JosephNaberhaus) [reply README.md:14]: Go 1.21 onwards will automatically install the toolchain specified in the `go.mod`, which is why that's the minimum.

I'm planning on rewording this anyway, so this doesn't need to block.

**Review by JosephNaberhaus — APPROVED** [review]
Looks good, thanks!

**Pull request merged**